admin/fetch_data.py

```diff
@@ -0,0 +1,160 @@
+"""
+Fetch reviews data from BigQuery and generate the CSV files used for these exercises.
+
+```sh
+python -m admin.fetch_data
+```
+"""
+
+from pathlib import Path
+
+import pandas as pd
+from google.cloud.bigquery import Client
+
+from recommender_systems.features import (
+    CustomerFeatures,
+    ProductFeatures,
+    ReviewFeatures,
+)
+from recommender_systems.splits import Splits
+
+SPLIT = "split"
+
+QUERY = f"""WITH
+    reviews AS (
+    SELECT
+        -- Reviews can be edited, or have their state changed, hence the need for these ANY_VALUE functions
+        ANY_VALUE(customer_id) AS customer_id_pii,
+        ANY_VALUE(tsin_id
+        HAVING
+            MAX updated_at) AS tsin,
+        uuid AS {ReviewFeatures.ID},
+        ANY_VALUE(rating
+        HAVING
+            MAX updated_at) AS {ReviewFeatures.RATING},
+        ANY_VALUE(body
+        HAVING
+            MAX updated_at) AS {ReviewFeatures.TEXT},
+        MAX(updated_at) AS {ReviewFeatures.TIMESTAMP},
+    FROM
+        `tal-production-ml.automated_review_moderation.reviews`
+    WHERE
+        moderation_state_id = 0 -- ACCEPTED
+        AND DATE(updated_at) BETWEEN "2025-01-01" AND "2025-07-01" -- RECENT
```

Comment:
Depending on how consistent you want this to be, maybe consider 


```suggestion
        AND DATE(updated_at) BETWEEN DATE_SUB(CURRENT_DATE, INTERVAL 180 DAY) AND CURRENT_DATE -- RECENT
```